mrp_production_request/models/mrp_production_request.py

```diff
@@ -0,0 +1,219 @@
+        digits_compute=dp.get_precision('Product Unit of Measure'),
+        readonly=True, states={'draft': [('readonly', False)]})
+    product_uom = fields.Many2one(
+        comodel_name='product.uom', string='Unit of Measure',
+        readonly=True, states={'draft': [('readonly', False)]},
+        domain="[('category_id', '=', category_uom_id)]")
+    category_uom_id = fields.Many2one(related="product_uom.category_id")
+    manufactured_qty = fields.Float(
+        string="Quantity in Manufacturing Orders",
+        compute=_compute_manufactured_qty, store=True,
+        digits_compute=dp.get_precision('Product Unit of Measure'),
+        readonly=True)
+    pending_qty = fields.Float(
+        string="Pending Quantity", compute=_compute_manufactured_qty,
+        store=True, digits_compute=dp.get_precision('Product Unit of Measure'),
+        readonly=True)
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom", string="Bill of Materials", required=True,
+        readonly=True, states={'draft': [('readonly', False)]})
+    routing_id = fields.Many2one(
+        comodel_name='mrp.routing', string='Routing',
+        on_delete='setnull', readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="The list of operations (list of work centers) to produce "
+             "the finished product. The routing is mainly used to compute "
+             "work center costs during operations and to plan future loads "
+             "on work centers based on production plannification.")
+    location_src_id = fields.Many2one(
+        comodel_name='stock.location', string='Raw Materials Location',
+        default=lambda self: self.env['stock.location'].browse(
+            self.env['mrp.production']._src_id_default()),
+        required=True, readonly=True, states={'draft': [('readonly', False)]})
+    location_dest_id = fields.Many2one(
+        comodel_name='stock.location', string='Finished Products Location',
+        default=lambda self: self.env['stock.location'].browse(
+            self.env['mrp.production']._dest_id_default()),
+        required=True, readonly=True, states={'draft': [('readonly', False)]})
+
+    @api.multi
+    def button_to_approve(self):
+        self.write({'state': 'to_approve'})
+        return True
+
+    @api.multi
+    def button_approved(self):
+        self.write({'state': 'approved'})
+        return True
+
+    @api.multi
+    def button_done(self):
+        self.write({'state': 'done'})
+        if self.mapped('procurement_id'):
+            self.mapped('procurement_id').write({'state': 'done'})
+        return True
+
+    @api.multi
+    def _check_reset_allowed(self):
+        if any([s in self._get_mo_valid_states() for s in self.mapped(
+                'mrp_production_ids.state')]):
+            raise UserError(
+                _("You cannot reset a manufacturing request with "
+                  "manufacturing orders not cancelled."))
```

Comment:
You cannot reset a manufacturing request if the related manufacturing orders are not canceled.